### ISSUE TITLE
Validate data ('dataset version' end point)

### ISSUE THREAD

**Issue by billgeo** [linz/geospatial-data-lake-team#4]
### User Story

In order to make sure my data is not corrupted, as a Topo Historic Imagery Data Maintainer, I want to validate my scanned historic imagery during dataset version creation.

#### Acceptance Criteria

- [x] Data is validated automatically against [mandatory checks
](https://confluence.linz.govt.nz/pages/viewpage.action?pageId=83655473)
- [ ] [Won't do] - ~~Validation status can be queried by the Data Maintainer~~
- [ ] [Won't do] - ~~Validation results are stored for later analysis by the Data Maintainer~~

#### How to test

1. Create a dataset in the data lake
1. Find a small local file you want to use as an asset, and get its hexadecimal SHA256 checksum ("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" is the checksum of the empty file if you don't have a way to get the SHA256 checksum). Add the string "1220" to the start of the checksum to create its multihash representation. So for the empty file the SHA256 multihash is "1220e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855".
1. Create a local metadata file with this content:
   ```json
   {
     "stac_version": "1.0.0-beta.2",
     "id": "dd6fb812bdb14341976794b21cb6abc7",
     "description": "any dataset description",
     "links": [],
     "license": "MIT",
     "extent": {
       "spatial": {
         "bbox": [
           [
             -180,
             -90,
             180,
             90
           ]
         ]
       },
       "temporal": {
         "interval": [
           [
             "1374-11-11T21:29:10+00:00",
             null
           ]
         ]
       }
     },
     "assets": {
       "any asset name": {
         "href": "URL",
         "checksum:multihash": "MULTIHASH"
       }
     }
   }
   ```
1. Replace `URL` in the above with the S3 URL of the file. This URL is on the form `s3://bucket-name/object-key`.
1. Replace `MULTIHASH` in the above with the calculated multihash.
1. If you want to test validation failure, you can add another asset file and metadata entry with just an arbitrary multihash. It should still be a hexadecimal string of the same length starting with "1220", so I would recommend just copying the valid multihash and just changing one of the characters after "1220" to an arbitrary hexadecimal character.
1. Upload the asset and metadata file to the same folder in S3, so that the asset URL corresponds to the one in the metadata file.
1. Give read access from the S3 bucket to the GDL.
1. Launch a dataset version creation with the following content template:

   ```json
   {
     "httpMethod": "POST",
     "body": {
       "id": "…",
       "metadata-url": "s3://…",
       "type": "RASTER",
     },
   }
   ```
1. When the step function finishes, go to the step function in AWS, click on check_files_checksums_single_task_batch_submit_job, then the batch job resource, then the log stream.
1. Verify that the expected log entries are in the list.

#### Tasks
- [x] Implement validate data task batch job for checksum validation.
   - Synopsis: `task.py --dataset-id=… --dataset-version=… --first-item=…`. 
- [x] Log validate data task arguments
- [x] Log checksum mismatch in validate data task including expected checksum, actual checksum, URL, "checksum mismatch" message
- [x] Return 0 from validate data task if successful or if a checksum mismatch was encountered
- [x] Implement [content iterator Lambda function](https://docs.aws.amazon.com/batch/latest/userguide/array_index_example.html) to loop over files (research!!)

#### Additional context

Detailed design: https://confluence.linz.govt.nz/display/LGDL/Dataset+Version+Endpoint
- For this user story (iteration 1) we will validate a single file at a time, and use AWS batch to check multiple in parallel.  
- Print nothing to standard output/standard error.

#### Definition of Ready
- [x] This story is __ready__ to work on, according to the [team's definition](https://confluence.linz.govt.nz/pages/viewpage.action?pageId=87930423)

#### Definition of Done
- [ ] This story is __done__, according to the [team's definition](https://confluence.linz.govt.nz/pages/viewpage.action?pageId=87930423)
